analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -118,15 +135,105 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
             {
```

Comment:
Unrelated to the PR:
The line above `objectCreation.GetFirstNonParenthesizedParent()` seems to be equivalent to `WalkUpParentheses` I created recently. Can you remove the "WalkUpParentheses" method and replace the invocation with "GetFirstNonParenthesizedParent". Thanks.

Reply:
I created #5790